hazelcast/src/main/java/com/hazelcast/internal/ascii/TextCommandServiceImpl.java

```diff
@@ -207,18 +205,11 @@ public long incrementTouchCount() {
 
     @Override
     public void processRequest(TextCommand command) {
-        if (responseThreadRunnable == null) {
-            synchronized (this) {
-                if (responseThreadRunnable == null) {
-                    responseThreadRunnable = new ResponseThreadRunnable();
-                    HazelcastThreadGroup hazelcastThreadGroup = node.getHazelcastThreadGroup();
-                    String threadNamePrefix = hazelcastThreadGroup.getThreadNamePrefix("ascii.service.response");
-                    Thread thread = new Thread(
-                            hazelcastThreadGroup.getInternalThreadGroup(), responseThreadRunnable, threadNamePrefix);
-                    thread.start();
-                }
-            }
-        }
+        HazelcastThreadGroup hazelcastThreadGroup = node.getHazelcastThreadGroup();
+        String threadNamePrefix = hazelcastThreadGroup.getThreadNamePrefix("ascii.service.response");
+        Thread thread = new Thread(
+                hazelcastThreadGroup.getInternalThreadGroup(), ResponseThreadRunnable.INSTANCE, threadNamePrefix);
+        thread.start();
```

Comment:
This isn't right - a thread will be spawned with each incoming request. This block ought to be moved into the enum ctor.